reducers/studyspacesReducer.js

```diff
@@ -45,7 +55,7 @@ export default (state = initialState, action = null) => {
       const newStudyspaces = ids.reduce(
         (spaces, s) =>
           updateStudyspaces(spaces, s, {
-            ...state.studyspaces.filter(x => x.id === s),
+            ...state.studyspaces.filter(x => x.id === s)[0],
```

Comment:
Same here too

Reply:
Yes if the studyspaces returned from the API change (e.g. some are removed/added).

But as with above, spreading the undefinedness inexplicably but conveniently produces an empty object.